src/xunit.netcore.extensions/Discoverers/ConditionalClassDiscoverer.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Xunit.NetCore.Extensions
+{
+    /// <summary>
+    /// This class discovers all of the tests and test classes that have
+    /// applied the ConditionalClass attribute
+    /// </summary>
+    public class ConditionalClassDiscoverer : ITraitDiscoverer
+    {
+        /// <summary>
+        /// Gets the trait values from the Category attribute.
+        /// </summary>
+        /// <param name="traitAttribute">The trait attribute containing the trait values.</param>
+        /// <returns>The trait values.</returns>
+        public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+        {
+            // If evaluated to false, skip the test class entirely.
+            if (!EvaluateParameterHelper(traitAttribute))
+            {
+                yield return new KeyValuePair<string, string>(XunitConstants.Category, XunitConstants.Failing);
+            }
+        }
+
+        internal static bool EvaluateParameterHelper(IAttributeInfo traitAttribute)
+        {
+            // Parse the traitAttribute. We make sure it contains two parts:
+            // 1. Type 2. nameof(conditionMemberName)
+            object[] conditionArguments = traitAttribute.GetConstructorArguments().ToArray();
+            Debug.Assert(conditionArguments.Count() == 2);
+
+            Type calleeType = null;
+            string[] conditionMemberNames = null;
+
+            if (ConditionalTestDiscoverer.CheckInputToSkipExecution(conditionArguments, ref calleeType, ref conditionMemberNames))
+            {
+                return true;
+            }
+
+            foreach (string entry in conditionMemberNames)
+            {
+                // Null condition member names are silently tolerated.
+                if (string.IsNullOrWhiteSpace(entry)) continue;
+
+                MethodInfo conditionMethodInfo = ConditionalTestDiscoverer.LookupConditionalMethod(calleeType, entry);
+                if (conditionMethodInfo == null)
```

Comment:
Can't we error, same as ConditionalTest:
```c#
                if ((conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
                {
                    return new[] 
                    {
                        new ExecutionErrorTestCase(
                            diagnosticMessageSink,
                            discoveryOptions.MethodDisplayOrDefault(),
                            testMethod,
                            GetFailedLookupString(conditionMemberName, declaringType))
                    };
                }
```

Could you make a follow up change? Otherwises tests will silently skip when they should not and we probably will never know.

Reply:
Sure, I will look into how XUnit reports error.

But at least we can do a `Debug.Assert()` here with a good message.